```
2024-08-08T07:20:28.4774229Z =================================== FAILURES ===================================
2024-08-08T07:20:28.4775470Z _ test_run[two_level_global_aggr-2-subscribe-and-recover-two-level-global-aggr] _
2024-08-08T07:20:28.4776038Z 
2024-08-08T07:20:28.4777290Z test_set = {'case_retried': 3, 'expected_results': [{'expected_results': [[100]], 'query_id': '3100'}, {'expected_results': [[101...:08:49.341101', 'query_id': 'c075b280-5554-11ef-8058-81034d3d5a33', 'query_id_type': 'non-designated', ...}, ...], ...}
2024-08-08T07:20:28.4778432Z caplog = <_pytest.logging.LogCaptureFixture object at 0x7f2a5534eb30>
2024-08-08T07:20:28.4778789Z 
2024-08-08T07:20:28.4778921Z     def test_run(test_set, caplog):
2024-08-08T07:20:28.4779263Z >       case_result_check(test_set)
2024-08-08T07:20:28.4779482Z 
2024-08-08T07:20:28.4779620Z test_stream_smoke/test_query_smoke.py:72: 
2024-08-08T07:20:28.4780074Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2024-08-08T07:20:28.4780384Z 
2024-08-08T07:20:28.4781243Z test_set = {'case_retried': 3, 'expected_results': [{'expected_results': [[100]], 'query_id': '3100'}, {'expected_results': [[101...:08:49.341101', 'query_id': 'c075b280-5554-11ef-8058-81034d3d5a33', 'query_id_type': 'non-designated', ...}, ...], ...}
2024-08-08T07:20:28.4782167Z logging_level = 'INFO'
2024-08-08T07:20:28.4782350Z 
2024-08-08T07:20:28.4782544Z     def case_result_check(test_set, logging_level="INFO"):
2024-08-08T07:20:28.4782954Z         #formatter = logging.Formatter(
2024-08-08T07:20:28.4783553Z         #    "%(asctime)s [%(levelname)8s] [%(processName)s] [%(module)s] [%(funcName)s] %(message)s (%(filename)s:%(lineno)s)"
2024-08-08T07:20:28.4784107Z         #)
2024-08-08T07:20:28.4784433Z         console_handler = logging.StreamHandler(sys.stderr)
2024-08-08T07:20:28.4784866Z         console_handler.formatter = formatter
2024-08-08T07:20:28.4785256Z         #logger.addHandler(console_handler)
2024-08-08T07:20:28.4785619Z         if logging_level=="INFO":
2024-08-08T07:20:28.4785955Z             logger.setLevel(logging.INFO)
2024-08-08T07:20:28.4786276Z         else:
2024-08-08T07:20:28.4786545Z             logger.setLevel(logging.DEBUG)
2024-08-08T07:20:28.4786866Z     
2024-08-08T07:20:28.4787159Z         expected_results = test_set.get("expected_results")
2024-08-08T07:20:28.4787659Z         statements_results_designed = test_set.get("statements_results_designed")
2024-08-08T07:20:28.4788075Z     
2024-08-08T07:20:28.4788318Z         if test_set["test_result"] != True:
2024-08-08T07:20:28.4788910Z             logger.info(f"expected_results = {expected_results}, statements_results_designed = {statements_results_designed}")
2024-08-08T07:20:28.4789448Z     
2024-08-08T07:20:28.4790003Z >       assert test_set["test_result"] == True, f"expected_result = {expected_results} \n, statements_results_designed = {statements_results_designed}"
2024-08-08T07:20:28.4791134Z E       AssertionError: expected_result = [{'query_id': '3100', 'expected_results': [[100]]}, {'query_id': '3100-1', 'expected_results': [[101]]}] 
2024-08-08T07:20:28.4792091Z E         , statements_results_designed = [{'query_id': '3100', 'query_result': [['100']]}, {'query_id': '3100-1', 'query_result': [['1'], ['100']]}]
2024-08-08T07:20:28.4792707Z E       assert False == True
2024-08-08T07:20:28.4792990Z E         +False
2024-08-08T07:20:28.4793239Z E         -True
2024-08-08T07:20:28.4793378Z 
2024-08-08T07:20:28.4793557Z test_stream_smoke/test_query_smoke.py:68: AssertionError
2024-08-08T07:20:28.4794072Z ------------------------------ Captured log call -------------------------------
2024-08-08T07:20:28.4795470Z INFO     test_query_smoke:test_query_smoke.py:66 expected_results = [{'query_id': '3100', 'expected_results': [[100]]}, {'query_id': '3100-1', 'expected_results': [[101]]}], statements_results_designed = [{'query_id': '3100', 'query_result': [['100']]}, {'query_id': '3100-1', 'query_result': [['1'], ['100']]}]
2024-08-08T07:20:28.4796840Z - generated html file: file:///actions-runner/_work/proton/proton/tests/stream/report_redp_2024-08-08 07:06:03.300431.html -
2024-08-08T07:20:28.4797505Z =========================== short test summary info ============================
2024-08-08T07:20:28.4798399Z FAILED test_stream_smoke/test_query_smoke.py::test_run[two_level_global_aggr-2-subscribe-and-recover-two-level-global-aggr]
2024-08-08T07:20:28.4799138Z ================== 1 failed, 849 passed in 862.53s (0:14:22) ===================
```

--- a/docker/test/proton_unit_test/Dockerfile
+++ b/docker/test/proton_unit_test/Dockerfile
@@ -1,5 +1,6 @@
 # docker build -t ghcr.io/timeplus-io/proton-unit-test .
-ARG FROM_TAG=latest
+ARG FROM_TAG=fae41f7606bb
+# https://github.com/timeplus-io/proton-enterprise/issues/5737
 FROM clickhouse/unit-test:$FROM_TAG
 
 COPY unit_tests_dbms /

--- a/tests/stream/test_stream_smoke/0030_two_level_global_aggr.yaml
+++ b/tests/stream/test_stream_smoke/0030_two_level_global_aggr.yaml
@@ -143,7 +143,7 @@ tests:
         - client: python
           query_type: table
           depends_on_stream: test_31_multishards_stream
-          wait: 1
+          wait: 2
           query: |
             insert into test_31_multishards_stream(i, v) values(101, 'test');
 
@@ -151,7 +151,7 @@ tests:
           query_type: stream
           query_id: 3100-1
           depends_on_stream: test_31_multishards_stream
-          wait: 1
+          wait: 2
           query: |
             recover from '3100';
 

